spec/requests/posts_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /posts' do
+    it 'returns http success' do
+      get '/users/index/posts'
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'returns http success' do
+      get '/users/index/posts/show'
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'renders the index template' do
+      get '/users/index/posts'
+      expect(response).to render_template('index')
+    end
+
+    it 'renders the show template' do
+      get '/users/index/posts/show'
+      expect(response).to render_template('show')
+    end
+
+    it 'includes the post title in the response' do
+      get '/users/index/posts'
+      expect(response.body).to include('Posts#index')
+    end
+
+    it 'includes the post body in the response' do
+      get '/users/index/posts/show'
+      expect(response.body).to include('Posts#show')
+    end
```

Comment:
- [ ] I would recommend writing two `describe` methods, one for `post#index` and another one for `post#show`. It is recommended to use different describe blocks for different methods and also it will make clear what we are expecting and what we are getting just by seeing the text. 


```ruby
RSpec.describe 'Posts', type: :request do
  describe 'GET /index' do
    ...
  end

  describe 'GET /show' do
    ...
  end
end
```